var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -250,17 +255,25 @@ def cmake_args(self):
             '-DLLVM_ENABLE_EH:BOOL=ON',
             '-DCLANG_DEFAULT_OPENMP_RUNTIME:STRING=libomp',
             '-DPYTHON_EXECUTABLE:PATH={0}'.format(spec['python'].command.path),
+            '-DLIBOMP_USE_HWLOC=On',
         ]
 
         projects = []
 
-        # TODO: Instead of unconditionally disabling CUDA, add a "cuda" variant
-        #       (see TODO above), and set the paths if enabled.
-        cmake_args.extend([
-            '-DCUDA_TOOLKIT_ROOT_DIR:PATH=IGNORE',
-            '-DCUDA_SDK_ROOT_DIR:PATH=IGNORE',
-            '-DCUDA_NVCC_EXECUTABLE:FILEPATH=IGNORE',
-            '-DLIBOMPTARGET_DEP_CUDA_DRIVER_LIBRARIES:STRING=IGNORE'])
+        if '+cuda' in spec:
+            cmake_args.extend([
+                '-DCUDA_TOOLKIT_ROOT_DIR:PATH=' + spec['cuda'].prefix,
+                '-DCUDA_NVCC_EXECUTABLE:FILEPATH=' + spec['cuda'].prefix.bin + '/nvcc',
```

Comment:
I believe this is redundant. You can just specify `-DCUDA_TOOLKIT_ROOT_DIR=...`

Reply:
I thought I tried that and it didn't work, but maybe that's because I didn't have the CUDAPackage base, giving that a try.